Dart/src/com/jetbrains/lang/dart/ide/findUsages/DartServerFindUsagesHandler.java

```diff
@@ -64,6 +66,19 @@ public boolean processInReadAction(SearchResult result) {
     return true;
   }
 
+  private static PsiElement findPsiElement(PsiElement context, Location location) {
+    final String locationFilePath = location.getFile();
+    final PsiFile psiFile = findPsiFile(context, locationFilePath);
+    if (psiFile != null) {
+      final int offset = location.getOffset();
+      PsiElement result;
+      if ((result = PsiTreeUtil.findElementOfClassAtOffset(psiFile, offset, DartReference.class, false)) != null) {
+        return result;
+      }
+    }
+    return null;
```

Comment:
Yes, unfortunately parser is not perfect in this area. I was told it is easy to fix, but so far I didn't do it. There's a special logic somewhere that finds doc comment for component, so while parser not fixed we need to write a helper that finds component for doc comment.